containers/src/Kokkos_DynamicView.hpp

```diff
@@ -173,7 +173,8 @@ struct ChunkedArrayManager {
     void execute() {
       // Destroy the array of chunk pointers.
       // Two entries beyond the max chunks are allocation counters.
-      for (unsigned i = 0; i < m_chunk_max; i++) {
+      auto const len = *reinterpret_cast<size_t*>(m_chunks + m_chunk_max + 1);
```

Comment:
```suggestion
      auto const len = static_cast<size_t>(m_chunks + m_chunk_max + 1);
```
also why do you even bother changing the type of the counter loop below?